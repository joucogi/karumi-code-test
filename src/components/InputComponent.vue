<template>
  <b-form-input :type="type" :placeholder="placeholder"></b-form-input>
</template>
<script>
export default {
  props: {
    type: {
      type: String,
      default: "text",
      required: false
    },
    placeholder: {
      type: String,
      default: "",
      required: false
    }
  }
};
</script>
<style scoped></style>
