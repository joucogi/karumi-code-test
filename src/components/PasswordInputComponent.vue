<template>
  <InputComponent
    :id="id"
    v-model="model"
    type="password"
    :placeholder="placeholder"
    @input="onInput"
  />
</template>
<script>
import InputComponent from "./InputComponent";

export default {
  components: { InputComponent },
  props: {
    id: {
      type: String,
      default: "",
      required: false
    },
    placeholder: {
      type: String,
      default: "",
      required: false
    },
    value: {
      type: String
    }
  },
  data() {
    return {
      model: this.value
    };
  },
  methods: {
    onInput(payload) {
      this.$emit("input", payload);
    }
  }
};
</script>
