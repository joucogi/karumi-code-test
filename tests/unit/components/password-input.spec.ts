import { shallowMount, Wrapper } from "@vue/test-utils";
import Vue from "vue";

import PasswordInputComponent from "@/components/PasswordInputComponent.vue";
import InputComponent from "@/components/InputComponent.vue";
import { checkInputProps, componentExists, toMatchSnapshot } from "../helpers";

let wrapper: Wrapper<Vue>;

beforeEach(() => {
  wrapper = shallowMount(PasswordInputComponent);
});

describe("PasswordInputComponent", () => {
  it("contains an InputComponent", () => {
    componentExists(wrapper, InputComponent);
  });

  it("shows an InputComponent with empty placeholder", () => {
    const input = wrapper.findComponent(InputComponent);
    checkInputProps(input, "", "password");
  });

  it("shows an InputComponent with Password text in placeholder", async () => {
    const placeholder = "Password";

    wrapper.setProps({
      placeholder
    });
    await wrapper.vm.$nextTick();

    const input = wrapper.findComponent(InputComponent);
    checkInputProps(input, placeholder, "password");
  });

  it("renders correctly", () => {
    toMatchSnapshot(wrapper);
  });
});
