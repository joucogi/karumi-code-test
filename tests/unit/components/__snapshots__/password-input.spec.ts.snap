// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PasswordInputComponent renders correctly 1`] = `
<inputcomponent-stub
  placeholder=""
  type="password"
/>
`;
